frontend_tests/casper_tests/10-admin.js

```diff
@@ -114,8 +114,9 @@ function submit_permissions_change() {
 casper.then(function () {
     casper.test.info("Test setting create streams policy to 'admins only'.");
     casper.waitUntilVisible("#id_realm_create_stream_policy", function () {
+        // by_admins_only
         casper.evaluate(function () {
-            $("#id_realm_create_stream_policy").val(settings_org.create_stream_policy_values.by_admins_only.code).change();
+            $("#id_realm_create_stream_policy").val(2).change();
```

Comment:
We could avoid the magic number with

```js
$("#id_realm_create_stream_policy option").filter(function () {
    return $(this).text() === "Admins";
}).prop("selected", true);
```

or

```js
$("#id_realm_create_stream_policy option:contains('Admins'):not(:contains('and'))").prop("selected", true);
```

Reply:
In the extremely likely event that these numbers change, I want this test to just break.

Reply:
(I prefer that to possibly false positives, plus it will be an easy fix.)

Reply:
I will revisit tomorrow.  I guess using strings here is a bit more robust.  Casper is kind of the bane of my existence, especially the webpack tracebacks.

Reply:
Yeah, I'm not sure whether it's worth the effort to try to make this nicer for Casper.  We're generally moving away from casper and I think there's a good chance we'll end up rewriting our frontend integration tests using a newer framework, so I don't want to spend a lot of time investing in tweaks to make constants like this more readable.

Reply:
FYI in case this comes up again as we deglobalize more modules, with #14021/#14023 I’ve implemented a more general solution in the form of `window.require`:

```js
const settings_config = window.require("./static/js/settings_config");
settings_config.create_stream_policy_values.by_admins_only.code
```